### ISSUE TITLE
Scaladoc eliding "case" for case objects

### ISSUE THREAD

**Issue by ddemange** [scala/bug#11871]
On the following simple code, 

```
package main

trait A
case class ClassA(i: Int) extends A
case object ObjectA extends A
```

scaladoc (version 2.12.3) is generating the following html output :
<img width="646" alt="image" src="https://user-images.githubusercontent.com/12810126/73428894-92e70800-433a-11ea-8048-8a78cdec0814.png">

The 'case' modifier is here for the class, but not for the object. Is there any reason for that?

**Comment by SethTisue** [comment]
just checking: is the problem present in Scala 2.13.1 as well?

**Comment by ddemange** [comment]
Yes, in the macos binary version 2.13.1, the problem is still here.

The output is also eliding the `extends A` on the class, which displays only when hovering on the class with the mouse.

<img width="596" alt="image" src="https://user-images.githubusercontent.com/12810126/73515433-94293b00-43f5-11ea-8549-7795aa6e0872.png">

**Comment by dwijnand** [comment]
(I understand now that "docs" is for documenting things in the project, and this issue is a "scaladoc tool" issue.)

**Comment by ddemange** [comment]
Thanks a lot for your prompt reaction.
I've just tested with the branch 2.13.x, and the 'case' is now displayed properly.

Any chance to back-port the fix to the 2.12.x branch?
I know it's getting old, but that's the version I'm depending on.

**Comment by SethTisue** [comment]
I believe a 2.12.x backport would be accepted if submitted.
